src/sssom/parsers.py

```diff
@@ -411,7 +411,9 @@ def from_sssom_dataframe(
     bad_attrs: typing.Counter[str] = Counter()
     for _, row in df.iterrows():
         mdict, bad_attrs = _get_mdict_ms_and_bad_attrs(row, bad_attrs)
-        mlist.append(_prepare_mapping(Mapping(**mdict)))
+        mapping = _ensure_valid_mapping_from_dict(mdict)
+        if mapping:
+            mlist.append(_prepare_mapping(mapping))
```

Comment:
maybe make the application of `_prepare_mapping` be a flaggable part of `_add_valid_mapping_to_list`

Reply:
Yes, and I refactored it also to make more explicit what this method does.